code/modules/mob/living/update_status.dm

```diff
@@ -49,7 +49,7 @@
 
 // Whether the mob is capable of talking
 /mob/living/can_speak()
-	return !(silent || (disabilities & MUTE) || is_muzzled())
```

Comment:
Instead of all the extra logic you're moving out of here, wouldn't it be better to make this proc a little bigger, and add the `muzzle.mute` check in here?